api/tests/e2e/api/test_inclusion_data.py

```diff
@@ -881,6 +881,90 @@ def test_can_filter_resources_by_sources(api_client, url, factory):
     assert_paginated_response_data(response.json(), total=0)
 
 
+@pytest.mark.with_token
+@pytest.mark.parametrize(
+    ("url"),
+    [
+        ("/api/v0/services"),
+        ("/api/v0/search/services"),
+        ("/api/v0/structures"),
+    ],
+)
+def test_filter_token_non_admin(api_client, url):
+    structure_1 = factories.StructureFactory(source="soliguide", code_insee="75056")
+    structure_2 = factories.StructureFactory(source="soliguide", code_insee="22001")
+    structure_3 = factories.StructureFactory(source="soliguide", code_insee="67270")
+    structure_4 = factories.StructureFactory(source="emplois-de-linclusion")
+
+    factories.ServiceFactory(structure=structure_1)
+    service_2 = factories.ServiceFactory(structure=structure_2)
+    service_3 = factories.ServiceFactory(structure=structure_3)
+    service_4 = factories.ServiceFactory(structure=structure_4)
+
+    response = api_client.get(
+        url,
+    )
+
+    assert response.status_code == 200
+    resp_data = response.json()
+    assert_paginated_response_data(resp_data, total=3)
+    if "services" in url:
+        assert {d["id"] for d in list_resources_data(resp_data)} == {
+            service_2.id,
+            service_3.id,
+            service_4.id,
+        }
+    else:
+        assert {d["id"] for d in list_resources_data(resp_data)} == {
+            structure_2.id,
+            structure_3.id,
+            structure_4.id,
+        }
+
+
+@pytest.mark.with_dora_token
+@pytest.mark.parametrize(
+    ("url"),
+    [
+        ("/api/v0/services"),
+        ("/api/v0/search/services"),
+        ("/api/v0/structures"),
+    ],
+)
+def test_filter_token_dora(api_client, url):
+    structure_1 = factories.StructureFactory(source="soliguide", code_insee="75056")
+    structure_2 = factories.StructureFactory(source="soliguide", code_insee="22001")
+    structure_3 = factories.StructureFactory(source="soliguide", code_insee="67270")
+    structure_4 = factories.StructureFactory(source="emplois-de-linclusion")
+
+    service_1 = factories.ServiceFactory(structure=structure_1)
+    service_2 = factories.ServiceFactory(structure=structure_2)
+    service_3 = factories.ServiceFactory(structure=structure_3)
+    service_4 = factories.ServiceFactory(structure=structure_4)
+
+    response = api_client.get(
+        url,
+    )
+
+    assert response.status_code == 200
+    resp_data = response.json()
+    assert_paginated_response_data(resp_data, total=4)
+    if "services" in url:
+        assert {d["id"] for d in list_resources_data(resp_data)} == {
+            service_1.id,
+            service_2.id,
+            service_3.id,
+            service_4.id,
+        }
+    else:
+        assert {d["id"] for d in list_resources_data(resp_data)} == {
+            structure_1.id,
+            structure_2.id,
+            structure_3.id,
+            structure_4.id,
+        }
```

Comment:
+1 pour la comparaison avec set

```suggestion
assert {d["id"] for d in list_resources_data(resp_data)} == {
        obj_1.id,
        obj_2.id,
        obj_3.id,
        obj_4.id,
    }
```